tests/interfaces/python/test_5_tcopflow.py

```diff
@@ -0,0 +1,40 @@
+import exago
+from mpi4py import MPI
+import tempfile
+import os
+import shutil
+import pytest
+from check_preconditions import check_preconditions
+import mpi4py.rc
+mpi4py.rc.threads = False
+
+check_preconditions()
+
+exago_ignore = -1000000
+
+
+def run_tcopflow(solver):
+    tcopf = exago.TCOPFLOW()
+    path = exago.prefix()
+
+    tcopf.set_tolerance(1.0E-03)
+    tcopf.set_network_data(os.path.join(
+        path, 'share', 'exago', 'datafiles', 'case9', 'case9mod_gen3_wind.m'))
+    tcopf.set_solver(solver)
+
+    tcopf.setup()
+    tcopf.solve()
+
+    assert tcopf.get_convergence_status()
+
+    obj = tcopf.get_objective()
+    assert isinstance(obj, float)
+
+    n = tcopf.get_num_iterations()
+    assert isinstance(n, int)
+
+
+@pytest.mark.nocomm
+@pytest.mark.MPI
+def test_tcopflow_1():
+    run_tcopflow('IPOPT')
```

Comment:
```suggestion
@pytest.mark.nocomm
@pytest.mark.MPI
def test_tcopflow_1():
    tcopf = exago.TCOPFLOW()
    path = exago.prefix()

    tcopf.set_tolerance(1.0E-03)
    tcopf.set_network_data(os.path.join(
        path, 'share', 'exago', 'datafiles', 'case9', 'case9mod_gen3_wind.m'))
    tcopf.set_solver('IPOPT')

    tcopf.setup()
    tcopf.solve()

    assert type(tcopf.get_convergence_status()) == bool

    obj = tcopf.get_objective()
    assert isinstance(obj, float)

    n = tcopf.get_num_iterations()
    assert isinstance(n, int)
```

I don't see the need to abstract our the `run_tcopflow` function here. In #140, make sure to name each test to be unique to the function being tested. Arguably, this function should be split up to test the usage of `get_convergence_status`, `get_objective` and `get_num_iterations` separately.

I know that our test cases are always going to converge, but I think asserting the function returns a bool is more sensible, and perhaps we also want a dummy problem that does not converge to validate we can return that correctly as well...

Reply:
It would be preferable to test each function individually. However, like with [sopflow](https://github.com/pnnl/ExaGO/blob/develop/tests/interfaces/python/test_4_sopflow.py) there are issues with the constructor/destructor. Simply creating a `tcopf` object then destroying it causes a seg fault bc some memory isn't allocated until `setup` is run. I suppose we could look into writing unit tests instead with calls to `setup` before testing each function but I opted to have us follow the sopflow framework first.

Reply:
Okay in digging through source code, it seems like we might need to re-visit `sopflow` python testing since we should be able to run `sopf = exago.SOPFLOW(); sopf.setup()` without causing too many issues... We should either fix this in an issue, or document it slightly better and explain...

For `tcopflow` though, I think writing tests such that `setup` is called isn't too bad. Perhaps there should also be an issue created though for fixing our destructors to not crash if people create and release an object without calling `setup`? Regardless we can figure that out in #140

Reply:
For more complex applications I think it makes sense to have an abstracted `run_tcopflow` to reduce copy and paste when we want to run a few times against a set of different configurations, but I think that can live alongside more "unit" tests while keeping the "functionality" tests.

Reply:
#142 and #140 should cover this